### PR TITLE
refactor(IAM): Limit permissions for firehose and cross account policies

### PR DESCRIPTION
**Pull request by BoboBananas**
## Summary

Review and limit IAM permissions for EKS Audit Logs Terraform template to adhere to least privilege
## How did you test this change?

https://lacework.atlassian.net/wiki/spaces/ENG/pages/2060877955/EKS+K8s+Audit+Ingestion+Design+Document#Data-Flow - Trimmed permissions using this to understand functionality of the different resources
Tested in dev3
Created resources using https://lacework.atlassian.net/wiki/spaces/ENG/pages/2172519599/Setup+Instructions#CloudFormation-script-1

## Issue
https://lacework.atlassian.net/browse/RAIN-58162

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -378,15 +378,10 @@ data "aws_iam_policy_document" "firehose_iam_role_policy" {
   statement {
     sid = "LaceworkEKSFirehoseIAMRole"
     actions = [
```
Comment (BoboBananas) [inline main.tf:380]: To transfer data from Firehose to S3, we only need PutObject

File: main.tf [hunk 2]
```diff
@@ -378,15 +378,10 @@ data "aws_iam_policy_document" "firehose_iam_role_policy" {
...
-      "s3:AbortMultipartUpload",
-      "s3:GetBucketLocation",
-      "s3:GetObject",
-      "s3:ListBucket",
-      "s3:ListBucketMultipartUploads",
       "s3:PutObject",
     ]
     effect    = "Allow"
```
Comment (BoboBananas) [inline main.tf:383]: Follows standard format for S3 bucket

File: main.tf [hunk 3]
```diff
@@ -451,9 +446,7 @@ data "aws_iam_policy_document" "eks_cross_account_policy" {
   statement {
     sid = "S3Permissions"
     actions = [
```
Comment (BoboBananas) [inline main.tf:448]: This is needed to read from S3 bucket, so we only need GetObject

File: main.tf [hunk 4]
```diff
@@ -465,23 +458,12 @@ data "aws_iam_policy_document" "eks_cross_account_policy" {
   statement {
     sid = "CloudWatchLogsPermissions"
     actions = [
```
Comment (BoboBananas) [inline main.tf:460]: Don't need permissions to update subscription filter

File: main.tf [hunk 5]
```diff
@@ -465,23 +458,12 @@ data "aws_iam_policy_document" "eks_cross_account_policy" {
...
-      "logs:DescribeSubscriptionFilters",
-      "logs:PutSubscriptionFilter"
+      "logs:DescribeSubscriptionFilters"
     ]
     effect    = "Allow"
     resources = [local.cloudwatch_permission_resources]
   }
```
Comment (BoboBananas) [inline main.tf:466]: Used to update subscription filter which isn't needed

**Review by afiune — APPROVED** [review]

**Pull request merged**